src/coreComponents/physicsSolvers/multiphysics/MultiResolutionHFSolver.cpp

```diff
@@ -0,0 +1,567 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file MultiResolutionHFSolver.cpp
+ *
+ */
+
+
+#include "MultiResolutionHFSolver.hpp"
+
+#include "common/GEOS_RAJA_Interface.hpp"
+#include "common/TimingMacros.hpp"
+#include "constitutive/ConstitutiveManager.hpp"
+#include "constitutive/contact/ContactSelector.hpp"
+#include "constitutive/fluid/SingleFluidBase.hpp"
+#include "discretizationMethods/NumericalMethodsManager.hpp"
+#include "finiteElement/Kinematics.h"
+#include "finiteVolume/FiniteVolumeManager.hpp"
+#include "finiteVolume/FluxApproximationBase.hpp"
+#include "mesh/DomainPartition.hpp"
+#include "mesh/SurfaceElementRegion.hpp"
+#include "mesh/MeshForLoopInterface.hpp"
+#include "mesh/utilities/ComputationalGeometry.hpp"
+#include "mesh/mpiCommunications/NeighborCommunicator.hpp"
+#include "physicsSolvers/contact/SolidMechanicsEFEMKernelsHelper.hpp"
+#include "physicsSolvers/contact/SolidMechanicsEmbeddedFractures.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.hpp"
+#include "physicsSolvers/surfaceGeneration/SurfaceGenerator.hpp"
+#include "physicsSolvers/surfaceGeneration/EmbeddedSurfaceGenerator.hpp"
+#include "linearAlgebra/utilities/LAIHelperFunctions.hpp"
+
+
+
+namespace geosx
+{
+
+namespace
+{
+//this function is not part of any class yet - just tries to find the number of a node given its coordinates
+// localIndex getNodeNumberFromCoordinate( MeshLevel const & mesh, real64 const & x, real64 const & y, real64 const & z)
+// {
+//   real64 tolerance = 1e-10;
+//   NodeManager const & meshNodeManager = mesh.getNodeManager();
+//   arrayView2d<real64 const> const nodalPositions = meshNodeManager.referencePosition();
+//   for (localIndex a=0; a < meshNodeManager.size(); a++)
+//     {
+//  real64 dist_sq = (nodalPositions( a,0 ) - x)*(nodalPositions( a,0 ) - x) + (nodalPositions( a,1 ) - y)*(nodalPositions( a,1 ) - y) +
+// (nodalPositions( a,2 ) - z)*(nodalPositions( a,2 ) - z);
+//  real64 dist = sqrt(dist_sq);
+//  if (dist < tolerance)
+//    {
+//      return a;
+//    }
+//     }
+//   return -1;
+// }
+
+//this function checks if an index is already in an array
+bool isMember( localIndex const & ind, array1d< localIndex > const & arr )
```

Comment:
this should go in LvArray if really necessary. Otherwise, would this work?

if( `std::find(arr.begin(), arr.end(), ind) != arr.end() `)